std/algorithm/iteration.d

```diff
@@ -2800,32 +2803,36 @@ if (isInputRange!RoR && isInputRange!(ElementType!RoR))
 
             private enum popBackEmptyElements = q{
                 // Skip over empty subranges.
-                if (_items.empty) goto end2;
-                while (_items.back.empty)
+                while (!_items.empty && _items.back.empty)
                 {
                     _items.popBack();
-                    if (_items.empty) goto end2;
                 }
-                checkFinalElement;
-                // We cannot export .save method unless we ensure subranges are not
-                // consumed when a .save'd copy of ourselves is iterated over. So
-                // we need to .save each subrange we traverse.
-                static if (isForwardRange!RoR && isForwardRange!(ElementType!RoR))
+                if (!_items.empty)
```

Comment:
This is sub-optimal as it will be executed one-more time than strictly necessary.

Reply:
Please see above comments